### PR TITLE
refactor: revamp test and build setup

### PR DESCRIPTION
**Pull request by KhaledSakr**
## What the PR Includes
- [x] Removed mocha/chai/sinon and replaced with jest/react-testing-library.
- [x] Replaced babel and gulp with tsc (for prod builds) and swc (for tests).
- [x] Rewrote tests in jest.

## Checklist
- [x] Tests are added.
- [ ] Manual testing instructions are added.
- [ ] Configs are added/adapted if applicable.
- [x] Issue is properly linked if applicable.

## Manual Testing Instructions
1. Testing pre-published version by making sure it installs correctly in clients.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+  transform: {
+    '^.+\\.(t|j)sx?$': '@swc/jest',
+  },
+  testEnvironment: 'jsdom',
+  testRegex: 'tests/.*test\\.tsx?$',
+  collectCoverage: true,
```
Comment (mowaleed96) [inline jest.config.js:10]: would it make sense to add coverage threshold so we make sure any changes added will have tests and coverage?

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  collectCoverageFrom: ['src/**/*.ts'],
```
Comment (mowaleed96) [inline jest.config.js:11]: should we add reporters 
```suggestion
  collectCoverageFrom: ['src/**/*.ts'],
  coverageReporters: ['html'],
```

File: jest.config.js (hunk 1 above)
Comment (KhaledSakr) [reply jest.config.js:10]: I think we need to integrate coveralls first and set a coverage threshold from there. I will add this issue to the DevOps backlog

**Review by mowaleed96 — APPROVED** [review]
LGTM :+1:

**Pull request merged**